src/main/scala/uk/gov/hmrc/integration/ServiceSpec.scala

```diff
@@ -35,33 +35,40 @@ trait ServiceSpec
   this: TestSuite =>
 
   override def fakeApplication(): Application =
-    GuiceApplicationBuilder(environment = Environment.simple(mode = applicationMode)).configure(configMap).build()
+    GuiceApplicationBuilder(environment = Environment.simple(mode = applicationMode.getOrElse(Mode.Dev)))
+      .configure(configMap)
+      .overrides(addtionalOverrides :_*)
+      .build()
 
   import uk.gov.hmrc.integration.UrlHelper._
 
   def externalServices: Seq[String]
 
   def additionalConfig: Map[String, _ <: Any] = Map.empty
 
+  def addtionalOverrides: Seq[GuiceableModule] = Seq.empty
+
   def testName: String = getClass.getSimpleName
 
-  def applicationMode: Mode.Value = Mode.Dev
+  def applicationMode: Option[Mode.Value] = Some(Mode.Dev)
+
+  def runModePrefix: String = applicationMode.map(m => s"${m.toString}.").getOrElse("")
```

Comment:
Are we introducing the concept of runMode?

If so, maybe this should be separate from applicationMode.
i.e. `def runMode: Option[String]` (or even `def runMode: Option[RunMode]` if dependent on bootstrap)
Then add a private `runModePrefix` to convert to string with `.`

If we're not, then maybe make private - since it's probably not obvious how to override.

Reply:
I have made it private, as don't think we need to introduce a new extension point